aws/resource_aws_codebuild_project_test.go

```diff
@@ -774,3 +793,87 @@ func testAccAWSCodeBuildProjectConfig_vpcConfig(subnets string) string {
   }
 `, subnets)
 }
+
+func testAccAWSCodebuildProjectConfig_buildBadgeUrlValidation(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_iam_role" "codebuild_role" {
+  name = "codebuild-role-%s"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "codebuild.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "codebuild_policy" {
+    name        = "codebuild-policy-%s"
+    path        = "/service-role/"
+    description = "Policy used in trust relationship with CodeBuild"
+    policy      = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Resource": [
+        "*"
+      ],
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_policy_attachment" "codebuild_policy_attachment" {
```

Comment:
Minor nitpick: We generally discourage the usage of `aws_iam_policy_attachment` in preference of `aws_iam_role_policy_attachment` in this case as the former resource often provides confusing/unexpected behavior for operators if they are unfamiliar with all the caveats of using that resource.